tests/pyfunc/test_chat_model.py

```diff
@@ -64,6 +64,9 @@ def predict(self, context, messages: List[ChatMessage], params: ChatParams) -> C
         mock_response = get_mock_response(messages, params)
         return ChatResponse(**mock_response)
 
+    def predict_stream(self, context, messages, params):
+        return iter([self.predict(context, messages, params)])
```

Comment:
Do we need to ask customers to write this code? Or it would work without that as well if someone extends the chat_model and just writes `predict` function.

Reply:
Customer doesn't need to write this code if they just want a default implementation of the predict_stream here https://github.com/mlflow/mlflow/blob/b9d9febf8812cb17ced528eb3c540f92acfe5b52/mlflow/pyfunc/loaders/chat_model.py#L82. But clearly this is not a `true` stream implementation as we're calling `predict` and simply convert the output to an iterator with a single item. I don't want to make `predict_stream` an abstract method, otherwise it'll break existing customers. But if user needs a real implementation of streaming, they need to override this function, and internally call something like `model.stream`